app/assets/javascripts/discourse/app/components/topic-map/private-message-map.gjs

```diff
@@ -0,0 +1,257 @@
+
+  @action
+  toggleEditing() {
+    this.isEditing = !this.isEditing;
+  }
+
+  <template>
+    <div class={{this.participantsClasses}}>
+      {{#each @postAttrs.allowedGroups as |group|}}
+        <PmMapUserGroup
+          @model={{group}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @removeAllowedGroup={{@removeAllowedGroup}}
+        />
+      {{/each}}
+      {{#each @postAttrs.allowedUsers as |user|}}
+        <PmMapUser
+          @model={{user}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @canRemoveSelfId={{@postAttrs.canRemoveSelfId}}
+          @removeAllowedUser={{@removeAllowedUser}}
+        />
+      {{/each}}
+    </div>
+
+    {{#if this.canShowControls}}
+      <div class="controls">
+        <DButton
+          @action={{this.actionAllowed}}
+          @label={{this.actionAllowedLabel}}
+          class="btn btn-default add-remove-participant-btn"
+        />
+
+        {{#if (and this.canInvite this.isEditing)}}
+          <DButton
+            @action={{@showInvite}}
+            @icon="plus"
+            class="btn btn-default no-text btn-icon add-participant-btn"
+          />
+        {{/if}}
+      </div>
+    {{/if}}
+  </template>
+}
+
+class PmMapUserGroup extends Component {
+  get canRemoveLink() {
+    return this.args.isEditing && this.args.canRemoveAllowedUsers;
+  }
+
+  get groupUrl() {
+    return groupPath(this.args.model.name);
+  }
+
+  <template>
+    <div class="user group">
+      <a href={{this.groupUrl}} class="group-link">
+        {{dIcon "users"}}
+        <span class="group-name">{{@model.name}}</span>
+      </a>
+    </div>
+    {{#if this.canRemoveLink}}
+      <PmRemoveGroupLink
+        @model={{@model}}
+        @removeAllowedGroup={{@removeAllowedGroup}}
+      />
+    {{/if}}
+  </template>
+}
+
+class PmRemoveGroupLink extends Component {
+  @service dialog;
+
+  @action
+  showConfirmDialog() {
+    this.dialog.deleteConfirm({
+      message: I18n.t("private_message_info.remove_allowed_group", {
+        name: this.args.model.name,
+      }),
+      confirmButtonLabel: "private_message_info.remove_group",
+      didConfirm: () => this.args.removeAllowedGroup(this.args.model),
+    });
+  }
+
+  <template>
+    {{! template-lint-disable no-invalid-interactive }}
+    <a
+      class="remove-invited no-text btn-icon btn"
+      {{on "click" this.showConfirmDialog}}
+    >
+      {{dIcon "times"}}
+    </a>
+  </template>
+}
+
+class PmMapUser extends Component {
+  @service site;
+
+  get linkClass() {
+    if (this.site.mobileView) {
+      return "";
+    }
+    return "user-link";
+  }
+
+  get userUrl() {
+    return this.args.model.path;
+  }
+
+  get avatarImage() {
+    return htmlSafe(
+      tinyAvatar(this.args.model.avatar_template, {
+        title: this.args.model.name || this.args.model.username,
+      })
+    );
+  }
```

Comment:
Instead of building the HTML manually, can we use the `{{avatar` helper directly in the template? I know @janzenisaac added some new params to it recently, and we can always add more if needed.